app/src/main/java/jet/EnemyJet.java

```diff
@@ -30,8 +36,42 @@ else if(h instanceof Bullet){
     }
 
     private JetLifeCycle jetLifeCycle;
-    public EnemyJet(float x, float y, float r, Paint p,int animationType) {
-        super(x,y,r,p,animationType);
+    public static class Builder extends Jet.Builder {
+        public Builder(float x, float y, float r,int animationType) {
+            super(x,y,r,animationType);
+        }
+        public Builder health(int val){
+            mHealth = val;
+            return this;
+        }
+
+        public Builder maxSpeed(int val){
+            mMaxSpeed = val;
+            return this;
+        }
+
+        public Builder hasDestination(boolean val){
+            mHasDestination = val;
+            return this;
+        }
+
+        public Builder setGunType(int val){
+            mGunType = val;
+            return this;
+        }
+
```

Comment:
Too many duplicate codes in builder. Should be able to put them into parent class. @Jelly-Yu